### DISCUSSION TITLE
Thoughts on expanded size variants for <Toast />

### DISCUSSION THREAD

**Discussion by andioneto** [twilio-labs/paste#834] [Ideas] [answered]
Hi folks 👋 
We have a page that allows users to bulk associate phone numbers to a specific SIP trunk instance.
- We know there are use cases where a customer has upward of 80,000 numbers associated to a specific trunk.
- We need to communicate to the user after the request has completed which numbers were successfully associated and which failed.
- We also need a good experience to handle a case where many numbers were requested and many failed (for example, 1000 failures, would create either 1000 toast or 1000 line items within a toast delineating failures.

We came up with this [POC demo](https://codesandbox.io/s/disclosing-error-info-93hkt?file=/src/App.js) that shows two options (one with Toast and one with Alert) that use a Disclosure to _disclose_ the error details.

We would like to use the Toast to consistently establish a pattern of user action leading to a toast, versus system/configuration error leading to an Alert, but since the width is set to be quite small, it makes the content very cramped.

Is this a potential case that might justify a few default toast widths? I'm thinking that within a set of Toasts on a single page, they would all need to be the same width. Curious to hear your thoughts 😸

**Comment by SiTaggart** [comment, marked as answer]
Hi,

My initial thought and reservation is that both alerts and toasts aren't really suitable for long form information consumption. They're really designed to give you a quick piece of feedback on what just happened, or in the case of an alert what the status is of a service or application might be.

One potential idea that I think you could explore is using a combination of a toast, and a modal which would be a better vehicle for more longer information. The toast could say "Success you associated 8000 numbers. <button>View details</button>". Upon clicking the button in the toast, you launch a dedicated modal with as much detailed information and room to digest it as is appropriate. If you don't care, you dismiss the toast.

Not sure what the rest of the team thinks about that. Will wait for further suggestions.

**Reply by andioneto** [reply]
Ok cool. That is very do-able as well, I'll run that by the folks working on the project. Thank you!

**Reply by aayushpi** [reply]
I like the modal approach too.

**Reply by serifluous** [reply]
+1 We're working on a proof of concept for notifications and feedback patterns. And this fits into the use case of using a Modal as a feedback indicator perfectly.

**Reply by serifluous** [reply]
If a customer does get 1000+ numbers that failed, what's their next step? Do they need to know exactly which numbers failed through the modal, or would they be able to understand that through the rest of the UI (for example, do they land on a table that tells them the status of each number?)

**Comment by andioneto** [comment]
Awesome, thank you both for the quick responses! 💯
